### ISSUE TITLE
React-router environment

### ISSUE THREAD

**Issue by sedubois** [graphcool-examples/react-relay-instagram-example#1]
[This](https://github.com/graphcool-examples/react-relay-instagram-example/blob/master/src/index.js#L26) example and others e.g the [todo example](https://github.com/graphcool-examples/react-relay-todo-example/blob/29ae10e9fb3ba10c11132d3efb34e11cdb364b7d/src/app.js#L25) specify a react-router "environment", but I can't find it neither in the react-router [API docs](https://github.com/ReactTraining/react-router/blob/v2.5.2/docs/API.md#props) nor in the [code](https://github.com/ReactTraining/react-router/blob/v2.5.2/modules/Router.js#L32). Could you explain the intention with this line and further documentation about it?

**Comment by schickling** [comment]
Hey @sedubois. Good point! The `environment` param is coming from [react-router-relay](https://github.com/relay-tools/react-router-relay). Do you think it would help to somehow mention that in the README?

**Comment by sedubois** [comment]
Thanks! I wonder how this works as `environment` is passed to `react-router`'s `Router`... (?) 
I'd say this is out of scope of graphcool, it's just the `react-router-relay` code logic which is confusing.

**Comment by schickling** [comment]
Yeah, I think that will change a lot anyway once `react-router` 4 lands kinda forcing a rewrite of `react-router-relay`.

**Comment by schickling** [comment]
@taion any insights on that you could share already? 👍

**Comment by taion** [comment]
There will be no react-router-relay for React Router v4. React Router v4 is set up in such a way that's incompatible with the way Relay wants to do data fetching across nested routes.

**Comment by schickling** [comment]
Are you aware of any best practises evolving how to use Relay with react-router 4?

**Comment by sedubois** [comment]
I think Relay 2 will get rid of routes, or did I dream that? Then I suppose that no such integration would be necessary any more.

**Comment by taion** [comment]
Your best bet with RRv4 is to use a `<Relay.Renderer>` at every route, like you would with mobile. This will lead to request waterfalls, though, which is especially bad in a web context.

I would recommend continuing to use RRv2 for the time being.
